pybaseball/statcast.py

```diff
@@ -1,175 +1,70 @@
 from datetime import date, timedelta
 from typing import List, Optional, Union
+import warnings
 
 import pandas as pd
 
 import pybaseball.datasources.statcast as statcast_ds
 
-from .utils import sanitize_date_range
+from .utils import sanitize_date_range, date_range
+from . import cache
 
 _SC_SINGLE_GAME_REQUEST = "/statcast_search/csv?all=true&type=details&game_pk={game_pk}"
 # pylint: disable=line-too-long
 _SC_SMALL_REQUEST = "/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C=&hfSea=&hfSit=&player_type=pitcher&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&game_date_gt={start_dt}&game_date_lt={end_dt}&team={team}&position=&hfRO=&home_road=&hfFlag=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details&"
 _MAX_SC_RESULTS = 40000
```

Comment:
doesn't seem like `_MAX_SC_RESULTS` is being used anymore? we should probably still catch the case where the query returns exactly 40000 rows because it's incomplete and basically a failure to complete the request

Reply:
Throw a highest level warning hoping that someone will report it as a GH issue and then we crank down our date chunking?
Parenthetical: the backoff was problematic, and the efficiency hit of asking, getting 40k, and then redoing it is severe. I am okay with not being fault tolerant in that sense where arguably it _should_ break. I don't see anything wrong with breaking on a breaking API change upstream. It's natural.

Reply:
I think we can assume we won't exceed the 40k and just remove the unused constant `_MAX_SC_RESULTS`

Reply:
I agree with @bdilday especially since we backed it to 7 days per chunk max.

Reply:
I kind of want to leave it in a comment?

Reply:
It's good to know / have there

Reply:
That's a good point. Should we just add a small check to this function where if length of results happens to hit this, we just throw an error or warning?